source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -375,6 +380,21 @@ json TransformerInstance::raw_string_callback(const inja::Arguments &args) const
   return val;
 }
 
+json TransformerInstance::replace_callback(const inja::Arguments& args) const {
+  const std::string& textToOperateOn = args.at(0)->get_ref<const std::string&>();
+  const std::string& regexToSearchFor = args.at(1)->get_ref<const std::string&>();
+  const std::string& stringToReplaceWith = args.at(2)->get_ref<const std::string&>();
+
+  try {
+    std::regex regex(regexToSearchFor);
```

Comment:
note to self: AFAIK currently this means we compile the regexes on the request path. We should find away around this